tests/OpenWeatherMapTest.php

```diff
@@ -144,6 +156,151 @@ public function testGetForecastUVIndex()
+                    '2f8796eefe67558dc205b09dd336d022',
+                    'Bundaberg',
+                    'AU',
+                    20
+                ),
+                null,
+                'AU',
+                20,
+            ],
+            [
+                'Bundaberg',
+                sprintf(
+                    'http://api.openweathermap.org/geo/1.0/direct?appid=%s&q=%s&limit=%d',
+                    '2f8796eefe67558dc205b09dd336d022',
+                    'Bundaberg',
+                    20
+                ),
+                null,
+                null,
+                20,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider coordinatesByLocationNameUrlDataProvider
+     */
+    public function testCorrectlyBuildsCoordinatesByLocationNameUrl(
+        $city,
+        $generatedUrl,
+        $stateCode = null,
+        $countryCode = null,
+        $limit = null
+    ) {
+        $requestFactory = $this->createMock(RequestFactory::class);
+        $requestFactory
+            ->expects($this->once())
+            ->method('createRequest')
+            ->with('GET', $generatedUrl);
+
+        $response = $this->createMock(ResponseInterface::class);
+        $response
+            ->method('getBody')
+            ->willReturn((new StreamFactory())->createStream('{}'));
+        $response
+            ->method('getStatusCode')
+            ->willReturn(200);
+        $httpClient = $this->createMock(Client::class);
+        $httpClient
+            ->expects($this->once())
+            ->method('sendRequest')
+            ->willReturn($response);
+
+        $this->owm = new OpenWeatherMap(
+            $this->apiKey,
+            $httpClient,
+            $requestFactory
+        );
+        $data = $this->owm->getCoordinatesByLocationName($city, $stateCode, $countryCode, $limit);
+    }
+
+    public function testRetrieveCoordinatesByLocationName()
+    {
+        $result = <<<EOF
+[{"name":"Bundaberg","local_names":{"ar":"بوندابيرج"},"zh":"班达伯格","en":"Bundaberg","ja":"バンダバーグ","ru":"Бундаберг","uk":"Бундаберг","lat":-24.8653253,"lon":152.3516785,"country":"AU","state":"Queensland"}]
+EOF;
+
+        $expected = [
+            new City (
+                -1,
+                'Bundaberg',
+                -24.8653253,
+                152.3516785,
+                "AU",
+                null,
+                null,
+                "Queensland",
+                [
+                    "ar" => "بوندابيرج",
+                    "en" => "Bundaberg",
+                    "ja" => "バンダバーグ",
+                    "ru" => "Бундаберг",
+                    "uk" => "Бундаберг",
+                    "zh" => "班达伯格",
+                ]
+            ),
+        ];
+
+        $mock = new MockHandler([
+            new Response(200, [], $result),
+        ]);
+
+        $handlerStack = HandlerStack::create($mock);
+        $httpClient = new Client(
+            new GuzzleClient(['handler' => $handlerStack])
+        );
```

Comment:
Can you use `TestHttpClient` from the tests folder instead of introducing a new dependency?